core/src/main/scala/org/bitcoins/core/crypto/Sign.scala

```diff
@@ -58,3 +59,14 @@ object Sign {
     }, publicKey)
   }
 }
+
+/** A signing interface for [[ExtKey]] */
+trait ExtSign extends Sign {
+
+  def deriveAndSignFuture: (ByteVector, BIP32Path) => Future[ECDigitalSignature]
+
+  /** First derives the child key that corresponds to [[BIP32Path path]] and then signs */
+  def sign(bytes: ByteVector, path: BIP32Path): ECDigitalSignature = {
+    Await.result(deriveAndSignFuture(bytes, path), 30.seconds)
```

Comment:
I don't think I like having this non-future version here, shouldn't the decision to block be left to the caller?

Reply:
This is following the pattern in `Sign`: https://github.com/bitcoin-s/bitcoin-s/blob/74f7f73797c04924994da0c6c1ee12e47aacbbfc/core/src/main/scala/org/bitcoins/core/crypto/Sign.scala#L28